implement several group methods

johnfreed opened the pull request:
These new methods come in both flavors: byte[] groupId and String base64GroupId.
- getGroupAdminMembers
- getGroupInviteUri
- getGroupPendingMembers
- getGroupRequestingMembers
- isAdmin
- isGroupAnnounceOnly
- setGroupAnnounceOnly
- updateAdmins
- updateMembers

An added version of updateGroup (the old one remains unchanged):
- updateGroup(groupId<ay>, name\<s\>, description\<s\>, addMembers\<s\>, removeMembers\<s\>, addAdmins\<s\>, removeAdmins\<s\>, resetGroupLink\<b\>, groupLinkState\<s\>, addMemberPermission\<s\>, editDetailsPermission\<s\>, avatar\<s\>, expiration\<i\>) -> groupId<ay>

sendTyping accepts either a string array of base64GroupId or a string array of recipient phone numbers
- sendTyping(typingAction\<b\>, base64GroupId<as>, recipients<as>) -> <>

AsamK commented:
I'm investigating an alternative approach to export each group as a separate dbus object with properties.

johnfreed commented:
OK thanks for the information. I have updated this pull request to remove the Base64 methods as you requested. I have also updated it to match the current master branch. As described in the log, the following are included here.

====
Reimplemented methods
====
removeDevice() -\> \<\>::
- becomes a device method (implemented by AsamK)

updateDeviceName(deviceName\<s\>) -\> \<\>::
- becomes setDeviceName and is a device method (implemented by AsamK)

setDeviceName(deviceName\<s\>) -\> \<\>::
- allow changing local name for device (updating AsamK's implementation)

getDevice(deviceId\<x\>) -\> devicePath\<o\>::
- prevent crash when negative number is given for deviceId (updating AsamK's implementation)

sendTyping(typingAction\<b\>, groupIdStrings\<as\>, numbers\<as\>) -\> \<\>::
- conform to sendTyping command and flesh it out

link() -\> deviceLinkUri\<s\>::
- restore variant where default name is "cli"

====
New device methods
====
getDeviceName() -\> deviceName\<s\>::

====
Already implemented, newly documented methods
====
getSelfNumber() -\> number\<s\>:: (implemented by AsamK)

====
Other new methods
====
getGroupId(groupName\<s\>) -\> groupId\<ay\>::
getGroupIdStrings() -\> groupIdStrings\<as\>::
getGroupIdString(groupName\<s\>) -\> groupIdString\<s\>::
getGroupNames() -\> groupNames\<as\>::
updateGroup(groupId\<ay\>, name\<s\>, description\<s\>, addMembers\<s\>, removeMembers\<s\>, addAdmins\<s\>, removeAdmins\<s\>, resetGroupLink\<b\>, groupLinkState\<s\>, addMemberPermission\<s\>, editDetailsPermission\<s\>, avatar\<s\>, expiration\<i\>, isAnnouncementGroup\<b\>) -\> groupId\<ay\>::
updateGroup(groupId\<ay\>, name\<s\>, description\<s\>, addMembers\<s\>, removeMembers\<s\>, addAdmins\<s\>, removeAdmins\<s\>, resetGroupLink\<b\>, groupLinkState\<s\>, addMemberPermission\<s\>, editDetailsPermission\<s\>, avatar\<s\>, expiration\<i\>) -\> groupId\<ay\>::
getGroupInviteUri(groupId\<ay\>) -\> inviteUri\<s\>::
getGroupAdminMembers(groupId\<ay\>) -\> groupAdminMembers\<as\>::
getGroupPendingMembers(groupId\<ay\>) -\> groupPendingMembers\<as\>::
getGroupRequestingMembers(groupId\<ay\>) -\> groupRequestingMembers\<as\>::
updateMembers(groupId\<ay\>, members\<as\>, addToMembers\<b\>) -\> memberList\<as\>::
updateAdmins(groupId\<ay\>, admins\<as\>, addToAdmins\<b\>) -\> adminList\<as\>::
setGroupAnnounceOnly(groupId\<ay\>, isAnnouncementGroup\<b\>) -\> \<\>::
isGroupAnnounceOnly(groupId\<ay\>) -\> isAnnouncementGroup\<b\>::
isAdmin(groupId\<ay\>) -\> isAdmin\<b\>::
